integration/container/run_test.go

```diff
@@ -0,0 +1,65 @@
+package container
+
+import (
+	"context"
+	"testing"
+
+	"github.com/docker/docker/client"
+	"github.com/docker/docker/integration/util"
+	"github.com/docker/docker/integration/util/request"
+	"github.com/docker/docker/pkg/parsers/kernel"
+)
+
+func daemonStorageDriver(ctx context.Context, cli client.APIClient) (string, error) {
+	info, err := cli.Info(ctx)
+	if err != nil {
+		return "", err
+	}
+	return info.Driver, nil
+}
+
+// #34573
+func TestRunVolumeLeakage(t *testing.T) {
+	defer setupTest(t)()
+	client := request.NewAPIClient(t)
+
+	// Note whether the test is running on devicemapper (where the error being
+	// tested ocurred). We run it anyway just to be safe.
+	driver, _ := daemonStorageDriver(context.Background(), client)
+	if driver != "devicemapper" {
+		t.Logf("Active driver (%s) is not devicemapper, continuing anyway.", driver)
```

Comment:
I think it would be a good measure to make sure the bug doesn't happen on other storage drivers too. I should probably just remove this check all-together.